microceph/common/fileUtils.go

```diff
@@ -0,0 +1,43 @@
+package common
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"github.com/canonical/lxd/shared/logger"
+	"github.com/djherbis/times"
+)
+
+// FilterFilesInDir filters filenames which matches the substring in path.
+func FilterFilesInDir(subString string, path string) []string {
+	files, err := os.ReadDir(path)
+	if err != nil {
+		return []string{}
+	}
+
+	resp := make([]string, len(files))
```

Comment:
This might be problematic -- as not all file names in dir will be added to the result we likely are going to end up with empty strings.